#!/usr/bin/env python

PACKAGE = "likelihood_grid"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add("leg1_x", double_t, 0, "X Coordinate of leg #1",    1.5, -10,   10)
gen.add("leg1_y", double_t, 0, "Y Coordinate of leg #1",    1.5, -10,   10)
gen.add("show_leg1",   bool_t,   0, "Show Leg 1",  True)

gen.add("leg2_x", double_t, 0, "X Coordinate of leg #2",    3.5, -10,   10)
gen.add("leg2_y", double_t, 0, "Y Coordinate of leg #2",    5.5, -10,   10)
gen.add("show_leg2",   bool_t,   0, "Show Leg 2",  True)

gen.add("leg3_x", double_t, 0, "X Coordinate of leg #3",    -3.5, -10,   10)
gen.add("leg3_y", double_t, 0, "Y Coordinate of leg #3",    -5.5, -10,   10)
gen.add("show_leg3",   bool_t,   0, "Show Leg 3",  True)

gen.add("torso_height", int_t, 0, "height of ROI", 120, 0, 480)
gen.add("torso_width", int_t, 0, "width of ROI", 160, 0, 640)
gen.add("torso_x", int_t, 0, "leftmost pixel of ROI", 200, 0, 1000)
gen.add("torso_y", int_t, 0, "topmost pixel of ROI", 125, 0, 1000)
gen.add("show_torso",   bool_t,   0, "Show Torso",  True)

gen.add("sound_pwr", double_t, 0, "sound source power", 30.0, 25.0, 50.0)
# gen.add("sound_x", double_t, 0, "sound x direction", 0.67, -1.0, 1.0)
gen.add("sound_y", double_t, 0, "sound y direction", -0.67, -1.0, 1.0)
gen.add("sound_dir", double_t, 0, "sound azimuth", -45.0, -180.0, 180.0)
gen.add("show_sound",   bool_t,   0, "Show Sound",  True)


exit(gen.generate(PACKAGE, "test", "Test"))
